File: 5kyu/PrimesinNumbers.py
def primeFactors(number):
    output,n="",number
    if n%2==0:
        power=0
        while n%2==0:n,power=n//2,power+1
        output += f"({2}**{power})" if power>1 else f"({2})"
    prime=3
    while n!=1:
        if n%prime==0:
            power=0
            while n%prime==0:n,power=n//prime,power+1
            output += f"({prime}**{power})" if power>1 else f"({prime})"
        prime+=2
    return output

File: 5kyu/PrimesinNumbers.cpp
#include <string>
class PrimeDecomp
{
public:
    static std::string factors(int num) {
      std::string output = "";

      if (num % 2 == 0) {
        output += removeFactors(num, 2);
      }

      int i = 3;
      while (num != 1) {
        if (num % i == 0) {
          output += removeFactors(num, i);
        }
        i += 2;
      }
      return output;
    }

    static std::string removeFactors(int &num, int factor) {
      std::string output = "(";
      output += std::to_string(factor);
      int power = 0;
      while (num % factor == 0) {
        num /= factor;
        power ++;
      }
      if (power > 1) {
        output += "**";
        output += std::to_string(power);
      }
      output += ")";
      return output;
    }

};

